server/routerlicious/packages/lambdas/src/scribe/checkpointManager.ts

```diff
@@ -48,21 +52,63 @@ export class CheckpointManager implements ICheckpointManager {
         //
         // And last we delete all mesages in the list prior to the summaryprotocol sequence number. From now on these
         // will no longer be referenced.
-        const dbOps = pending.map((message) => ({ ...message,
-            mongoTimestamp: new Date(message.operation.timestamp) }));
-        if (dbOps.length > 0) {
-            await runWithRetry(
-                async () => this.opCollection.insertMany(dbOps, false),
-                "writeCheckpointScribe",
-                3 /* maxRetries */,
-                1000 /* retryAfterMs */,
-                getLumberBaseProperties(this.documentId, this.tenantId),
-                (error) => error.code === 11000 /* shouldIgnoreError */,
-                (error) => !this.clientFacadeRetryEnabled, /* shouldRetry */
-            );
+        if (!this.getDeltasViaAlfred) {
```

Comment:
>!this.getDeltasViaAlfred [](http://example.com/codeflow?start=12&length=24)

(nit) swap the two branches of the if statement to avoid the not operator.

Reply:
> > ```
> >     let document: IDocument;
> > ```
> 
> (Unrelated to this change, no action suggested) this is a long function 😄
> 
> Refers to: server/routerlicious/packages/lambdas/src/scribe/lambdaFactory.ts:70 in [9f3acc7](https://github.com/microsoft/FluidFramework/commit/9f3acc74a3b641eb8d4916cb49d419e985e07f07). [](commit_id = [9f3acc7](https://github.com/microsoft/FluidFramework/commit/9f3acc74a3b641eb8d4916cb49d419e985e07f07), deletion_comment = False)

I agree. Lots of scope for cleanup here. From our logs, it looks like most of line 134-159 is not getting exercised at all. Will create a task to track this and the other scribe cleanup tasks.